src/mlte/spec/spec.py

```diff
@@ -5,24 +5,25 @@
 from __future__ import annotations
 
 import time
-from itertools import groupby, combinations
-from typing import List, Dict, Iterable, Any, Union
+from itertools import groupby
+from typing import Iterable, Any, Union, Type, Optional
 
 from mlte.property import Property
-from mlte.measurement.validation import ValidationResult
+from mlte.validation import Result
 from mlte._private.schema import SPEC_LATEST_SCHEMA_VERSION
 from mlte._global import global_state
 from mlte.api import read_spec, write_spec
-from mlte.binding import Binding
 from .bound_spec import BoundSpec
+from .condition import Condition
+from mlte.measurement import Measurement
 
 
-def _unique(collection: List[str]) -> bool:
+def _unique(collection: list[str]) -> bool:
```

Comment:
Starting from Python 3.9, "list" can be used as the type for type hints, instead of List. Same for dict and others. And to get this to work on 3.8, just adding "from __future__ import annotations" makes it work the same way. Since using list and dict instead of List and Dict is the way forward for Python type hints, I am trying to switch to that usage when possible. See https://stackoverflow.com/a/65777001

Reply:
Great! Moving us into the next generation.